Gui/opensim/view/src/org/opensim/view/markerEditor/NewMarkerAction.java

```diff
@@ -77,18 +77,19 @@ public void actionPerformed(ActionEvent evt) {
         Model model = markersNode.getModelForNode();
         Vec3 offset = new Vec3(0.11, 0.22, 0.33);
         MarkerSet markerset = model.getMarkerSet();
-        Body body = model.getBodySet().get(0);
-        if (body == null) {
-            return;
-        }
+        PhysicalFrame frameToUse;
+        if ( model.getBodySet().getSize()==0)
+            frameToUse = model.get_ground(); // model only has Ground
+        else// Model has at least one body in BodySet.
+            frameToUse = model.getBodyList().begin().__deref__();
```

Comment:
Why not always use ground?

Reply:
Maintain old behavior but possible to always use Ground

Reply:
Ground will not be in the BodySet, so I think this actually changes behavior.

Reply:
The old code would actually crash if there are no Bodies in the model altogether, so this was a defensive change, but I agree that creating the new Marker in ground always would be more consistent with old behavior. Will do :+1